lib/blocks.php

```diff
@@ -102,7 +103,7 @@ function do_blocks( $content ) {
 	global $wp_registered_blocks;
 
 	// Extract the blocks from the post content.
-	$matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)\s*\/?-->(?:.*?<!--\s*\/wp:\g1\s+-->)?/';
+	$matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*?)\s*\/?-->(?:.*?<!--\s*\/wp:\g1\s+-->)?/';
```

Comment:
@dmsnell The reason for this is the matches for a block like `<!-- wp:core/latestposts {"poststoshow":3} /-->`. Given that example, without the change on this line it was resulting in a match like `{"poststoshow":3} /`, which naturally is invalid JSON. So I was debating whether to `rtrim` the `/` or to make the pattern matching non-greedy. I decided on the latter, but you think the former is better then I'm happy to change.